### PR TITLE
Pagination and Total Results

### PR DESCRIPTION
**Pull request by mjoyce91**
Implements #261 and #299 by incorporating [react-paginate](https://github.com/AdeleD/react-paginate) to perform pagination, and including a component to display "Viewing # - # of # results" on the results page.

### CODE REVIEW

File: src/Components/TotalResults/TotalResults.test.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  it('matches snapshot', () => {
+    wrapper = shallow(
+      <TotalResults total={total} pageNumber={pageNumber} pageSize={pageSize} />,
+    );
+    expect(toJSON(wrapper)).toMatchSnapshot();
+  });
+
+  it('displays pager numbers and total results correctly when reaching the second to last page of the results', () => {
+    wrapper = mount(
```
Comment (burgwyn) [inline src/Components/TotalResults/TotalResults.test.jsx]: Can we test without `mount`?

File: src/Components/TotalResults/TotalResults.test.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  let wrapper = null;
+
+  const middlewares = [thunk];
+  const mockStore = configureStore(middlewares);
+
+  const total = 103;
+  const pageNumber = 4;
+  const pageSize = 25;
+
+  it('can receive props', () => {
```
Comment (burgwyn) [inline src/Components/TotalResults/TotalResults.test.jsx]: Let's test with zeroes

File: src/Components/TotalResults/__snapshots__/TotalResults.test.jsx.snap [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (burgwyn) [inline src/Components/TotalResults/__snapshots__/TotalResults.test.jsx.snap]: Let's do a few more snapshots - page 1, 2, n-1, n

File: src/Containers/Results/Results.jsx [hunk 4]
```diff
@@ -17,22 +17,38 @@ class Results extends Component {
       query: { value: window.location.search.replace('?', '') || '' },
       defaultSort: { value: '' },
       defaultPageSize: { value: '' },
+      defaultPageNumber: { value: 1 },
```
Comment (burgwyn) [inline src/Containers/Results/Results.jsx]: Should this be 0?

File: src/Containers/Results/Results.jsx [hunk 5]
```diff
@@ -68,6 +84,7 @@ class Results extends Component {
           defaultSort={this.state.defaultSort.value}
           pageSizes={POSITION_PAGE_SIZES}
           defaultPageSize={this.state.defaultPageSize.value}
+          defaultPageNumber={this.state.defaultPageNumber.value - 1}
```
Comment (burgwyn) [inline src/Containers/Results/Results.jsx]: This seems weird

File: src/Components/TotalResults/TotalResults.test.jsx (hunk 1 above)
Comment (mjoyce91) [inline src/Components/TotalResults/TotalResults.test.jsx]: Ah, for some reason I thought I needed `mount` to do this. `shallow` works though.

File: src/Containers/Results/Results.jsx (hunk 4 above)
Comment (mjoyce91) [inline src/Containers/Results/Results.jsx]: The pagination module is zero-based, but the API pagination starts at 1.

File: src/Containers/Results/Results.jsx (hunk 5 above)
Comment (mjoyce91) [inline src/Containers/Results/Results.jsx]: Maybe there's a better place to handle the reconciliation between the pagination module being zero-based and the API starting with `page=1`.

File: src/Components/TotalResults/TotalResults.test.jsx (hunk 2 above)
Comment (mjoyce91) [inline src/Components/TotalResults/TotalResults.test.jsx]: Updated to use zeroes for the generic tests

File: src/Containers/Results/Results.jsx (hunk 5 above)
Comment (burgwyn) [inline src/Containers/Results/Results.jsx]: Let's make an issue for the reconciliation

File: src/Components/TotalResults/__snapshots__/TotalResults.test.jsx.snap (hunk 3 above)
Comment (mjoyce91) [inline src/Components/TotalResults/__snapshots__/TotalResults.test.jsx.snap]: Beat me with merge before I could add this 😀

**Pull request merged**